### ISSUE TITLE
AccessDenied running ps.pgrep on Windows Minion

### ISSUE THREAD

**Issue by Tamerz** [saltstack/salt#20475]
When I run:
`sudo salt 'wintest' ps.pgrep notepad`
No matter what I choose for the pattern, I get:
```
wintest:
    The minion function caused an exception: Traceback (most recent call last):
      File "salt/minion.py", line 1022, in _thread_return
      File "salt/modules/ps.py", line 292, in pgrep
      File "salt/modules/ps.py", line 67, in _get_proc_name
      File "psutil/__init__.py", line 490, in name
      File "psutil/_pswindows.py", line 206, in wrapper
      File "psutil/_pswindows.py", line 241, in name
      File "psutil/_pswindows.py", line 213, in wrapper
    AccessDenied: (pid=224)
```
On the minion, PID 224 is "smss.exe", "Windows Session Manager".

Both the master and the minion are running v2014.7.1

The minion is running as the local system account (default).

**Comment by UtahDave** [comment]
Thanks! We'll investigate.

**Comment by eliasp** [comment]
This is a "known" issue and 2 months ago I started working on it, but never got around to complete + test it.

This happens, because the details of some processes returned by `psutil.process_iter()` can't be accessed due to permission issues. Those processes should be simply skipped by properly handling the returned exceptions when processing the list returned by `psutil.process_iter()`.

See also: http://stackoverflow.com/a/19731902/2858390

**Comment by eliasp** [comment]
Completely missed that I already filed an issue for this: #18689 
Fix is available there, but still needs some more testing/cleanup/generalization before I'll turn this into a PR.

So I think this issue can be closed as dup of #18689

**Comment by eliasp** [comment]
Fix available in PR#20510.

@Tamerz Could you give [the fixed salt/modules/ps.py]() a try whether it fixes the issue for you?
To test it, you can use [Salt's Dynamic Module Distribution](http://docs.saltstack.com/en/latest/ref/file_server/dynamic-modules.html) to distribute the updated version of `ps.py` for testing/until the next release.

**Comment by Tamerz** [comment]
@eliasp Thanks, I tried that out on my test machine and it appears to have resolved my issue.
